### ISSUE TITLE
_ensureIndex on more fields

### ISSUE THREAD

**Issue by mitar** [vsivsi/meteor-job-collection#122]
Should we have indexes also on:

```
collection._ensureIndex
  priority: 1
  retryUntil: 1
  after: 1
```

`getWork` sorts by those fields.

I have not measured anything, this is just by observing queries.

**Comment by vsivsi** [comment]
Probably yes...  I'm still living in a mindset where indexing is a developer decision, but I guess I already violated that some time ago.  I don't have much time between now and the end of the year, so this kind of thing (not really a bug) will probably need to wait.

**Comment by mitar** [comment]
Should we add this then just to the documentation? To the section where we are mentioning indexes?

**Comment by rhyslbw** [comment]
+1 for just adding them in. Is there anytime these wouldn't want to be indexed?

**Comment by vsivsi** [comment]
@mitar  If you want to implement this, I'm fine with it.

**Comment by mitar** [comment]
Fixed in `dev` branch.

**Comment by rhyslbw** [comment]
:+1: FYI @vsiverson and @mitar I'm about to look into implementing Job Collection as a work queue for management tasks in [Space.eventSourcing](https://github.com/meteor-space/event-sourcing), such as snapshotting and projection rebuilds, so I'll be around if you need to bounce any ideas etc

**Comment by aldeed** [comment]
@mitar @vsivsi It seems like a single key index on `status` is also needed. This is currently listed as the slowest query on our Compose:

<img width="768" alt="screenshot 2015-12-02 08 46 58" src="https://cloud.githubusercontent.com/assets/3012067/11533896/a226e122-98d1-11e5-91da-01786efe755f.png">

Using package version 1.2.2

**Comment by mitar** [comment]
Where does this query comes from?

**Comment by aldeed** [comment]
Oh good point, looks like it's from some code we have that checks for stuck jobs periodically. So I guess it's up to you whether you want the package to set up the index. I believe pkgs should ensure indexes for the queries they do, but whether you want to anticipate other common queries, that's personal preference I guess.

**Comment by mitar** [comment]
Yea, I think this should be done by your code then. I think this is clear that if you are doing some other queries it is on you to create indexes.

**Comment by mitar** [comment]
I see that [we are also doing queries on status](https://github.com/mitar/council-app/blob/master/packages/classy-job/worker.coffee#L73) for finding stale jobs. But I do create [our own indexes](https://github.com/mitar/council-app/blob/master/packages/classy-job/worker.coffee#L73).

**Comment by mitar** [comment]
Hm, but those are no extra indexes, besides what is provided by this package.

**Comment by aldeed** [comment]
I don't think the type-status combined index helps when you're querying on status only

**Comment by mitar** [comment]
> I don't think the type-status combined index helps when you're querying on status only

That is my understanding as well.

**Comment by vsivsi** [comment]
I agree that the package should only create indexes for queries it generates internally.  Maintaining indexes isn't free, so generating indexes that you don't know for will be needed is wasteful.

**Comment by vsivsi** [comment]
Mitar's additional indexes will be released in 1.3.0 on Atmosphere today.
